Content.Shared/Physics/ClimbController.cs

```diff
@@ -0,0 +1,84 @@
+﻿#nullable enable
+using Robust.Shared.Maths;
+using Robust.Shared.Physics;
+
+namespace Content.Shared.Physics
+{
+    /// <summary>
+    /// Movement controller used by the climb system. Lerps the player from A to B.
+    /// Also does checks to make sure the player isn't blocked.
+    /// </summary>
+    public class ClimbController : VirtualController
+    {
+        private Vector2? _movingTo = null;
+        private Vector2 _lastKnownPosition = default;
+        private int _numTicksBlocked = 0;
+
+        /// <summary>
+        /// If 5 ticks have passed and our position has not changed then something is blocking us.
+        /// </summary>
+        public bool IsBlocked => _numTicksBlocked > 5 || _isMovingWrongDirection;
+
+        /// <summary>
+        /// If the controller is currently moving the player somewhere, it is considered active.
+        /// </summary>
+        public bool IsActive => _movingTo.HasValue;
+
+        private float _cancelTime = default;
+        private float _initialDist = default;
+        private bool _isMovingWrongDirection = false;
+
+        public void TryMoveTo(Vector2 from, Vector2 to)
+        {
+            if (ControlledComponent == null)
+            {
+                return;
+            }
+
+            _initialDist = (from - to).Length;
+            _numTicksBlocked = 0;
+            _lastKnownPosition = from;
+            _movingTo = to;
+            _isMovingWrongDirection = false;
+        }
+
+        public override void UpdateAfterProcessing()
+        {
+            base.UpdateAfterProcessing();
+
+            if (ControlledComponent == null || _movingTo == null)
+            {
+                return;
+            }
+
+            if (ControlledComponent.Owner.Transform.WorldPosition.EqualsApprox(_lastKnownPosition, 0.01))
```

Comment:
Same as the change on line 61, to make the check more lenient.

```suggestion
            if (ControlledComponent.Owner.Transform.WorldPosition.EqualsApprox(_lastKnownPosition, 0.05))
```